app/controllers/peptidome_controller.rb

```diff
@@ -47,6 +47,20 @@ def get_sequences
     render json: Oj.dump(Sequence.list_sequences(ids).join("\n"), mode: :compat)
   end
 
+  # Returns a list of proteins
+  def get_proteins
+    ids = ProteomeCache.delta_decode(JSON(params[:sequence_ids]))
+    sequences = Sequence.includes(original_peptides: :uniprot_entry).where(id: ids)
+    data = Hash[sequences.map { |sequence| [sequence, sequence.original_peptides.map(&:uniprot_entry)] }]
```

Comment:
As data isn't used as a `Hash` anywhere, wouldn't it be faster to just keep it as a list?

``` ruby
    data = sequences.map { |sequence| [sequence, sequence.original_peptides.map(&:uniprot_entry)] }
```


_[Original comment](https://github.ugent.be/unipept/unipept/pull/576#discussion_r4515) by @ninewise on Tue Jun 07 2016 at 09:59._